### ISSUE TITLE
Include a helio centric velocity correction

### ISSUE THREAD

**Issue by profxj** [PYPIT/PYPIT#79]
As titled.

**Comment by rcooke-ast** [comment]
heliocentric correction is already implemented. It is in the function `arvcorr.helio_corr()`.
The present implementation makes updates to the wavelength IDs, and then rederives the wavelength solution. Therefore, the heliocentric correction should be done before extraction, but after the flexure shift.

In fact, this is on line 206 of `armlsd.py`, but has since become deprecated, since the extraction step is performed earlier. This line of code should therefore be moved into `arproc.reduce_frame()`

As a final note, we should eventually implement a full barycentric velocity correction. I'm adding that as a separate issue now...

**Comment by profxj** [comment]
The current flexure implementation requires an extraction of the sky (i.e. after extraction).

Would it be ok to apply to the extracted wavelengths (our LowRedux approach, for
what it is worth)?

**Comment by rcooke-ast** [comment]
I see...

It may not be far wrong, and we can do that temporarily. Eventually, both the flexure correction and the heliocentric correction should just shift the wavelengths in the arc id solution (to save continuously perturbing the wavelength solution or shifting the pixels).

Also, I think it's formally incorrect to do the heliocentric correction after the extraction on long slit data. The extraction should be mapped onto pixels of constant delta wavelength (for single order spectroscopy) or constant velocity (for echelle)... right? ... so that the FWHM resolution (which is either constant in delta wavelength or delta velocity, respectively) is sampled by ~2-3 pixels. It is therefore unwise to perform the heliocentric correction (a velocity correction) on pixels of constant delta wavelength.

Does this make sense, or am I mistaken?

**Comment by profxj** [comment]
> On May 17, 2016, at 4:36 PM, Ryan Cooke <notifications@github.com> wrote:
> 
> I see...
> 
> It may not be far wrong, and we can do that temporarily. Eventually, both the flexure correction and the heliocentric correction should just shift the wavelengths in the arc id solution (to save continuously perturbing the wavelength solution or shifting the pixels).
> 
> 
We should talk more about this.  I tried implementing it for the flexure
and was not happy with the results and was also concerned it wasn’t
the right approach anyhow.  I think the same issues apply for the
heliocentric.
> Also, I think it's formally incorrect to do the heliocentric correction after the extraction on long slit data. The extraction should be mapped onto pixels of constant delta wavelength (for single order spectroscopy) or constant velocity (for echelle)... right? ... so that the FWHM resolution (which is either constant in delta wavelength or delta velocity, respectively) is sampled by ~2-3 pixels. It is therefore unwise to perform the heliocentric correction (a velocity correction) on pixels of constant delta wavelength.
> 
> 
We are not mapping to constant anything so far.  And might not.
So we are (nearly) preserving the FWHM and sampling.
> Does this make sense, or am I mistaken?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/PYPIT/PYPIT/issues/79#issuecomment-219885365>
> !DSPAM:826,573baaeb46521834017520!

----------------------------------------------
J. Xavier Prochaska
UCO/Lick Observatory
1156 High St.
UC Santa Cruz
Santa Cruz, CA 95064

http://www.ucolick.org/~xavier/
http://imps.ucolick.org/
http://www.qpqsurvey.org/
http://k1dm3.ucolick.org/
http://www.ucolick.org/~xavier/IDL/index.html
https://github.com/pyigm/pyigm
https://github.com/linetools/linetools

831-459-2135 (Direct)
831-459-2991 (UCO/Lick Main)
831-459-5244 (Fax)

**Comment by rcooke-ast** [comment]
Implemented in PR #190
